src/Bufferer/PipeBufferer.php

```diff
@@ -14,38 +13,43 @@ class PipeBufferer implements BuffererInterface
 {
     private $logger;
     private $inputStream;
-    private $outputStream;
 
     private $mimeType;
     private $filePath;
     private $progressBar;
 
     private $buffering = true;
 
-    /**
-     * @param resource $inputStream
-     */
+    private $deferred;
+
     public function __construct(
         LoggerInterface $logger,
-        $inputStream,
+        InputStream $inputStream,
         ?string $outputPath,
         ConsoleSectionOutput $output
     ) {
         $this->logger = $logger;
-        $this->inputStream = new ResourceInputStream($inputStream);
-        $this->outputStream = new ResourceOutputStream($fileOutput = $outputPath ? fopen($outputPath, 'w') : tmpfile());
+        $this->inputStream = $inputStream;
         $this->mimeType = new \Amp\Deferred;
-        $this->filePath = $outputPath ?: stream_get_meta_data($fileOutput)['uri'];
+        $this->filePath = $outputPath ?? \tempnam(\sys_get_temp_dir(), 'stdinho') . '.tmp';
         $this->progressBar = new ProgressBar($output, 0, 'buffer');
     }
 
     public function __invoke(): \Generator
     {
         $this->logger->debug("Saving stdin to $this->filePath");
 
+        $outputStream = yield \Amp\File\open($this->filePath, 'wb');
```

Comment:
Assuming you don't have extensions installed, there will probably be some decrease in performance from blocking functions due to the IPC that parallel uses. However, it's the only way to be non-blocking. Every non-blocking implementation I'm aware of uses processes/threads for file handling. Node.js uses libuv, which of course PHP can use through the uv extension. Getting rid of amphp/file will affect concurrency.

If you're noticing bugs we'd appreciate bug reports. I noticed a warning from eio when writing an empty string that I fixed today. Otherwise I've been using eio in production without issues.

Reply:
As I mentioned in https://github.com/amphp/byte-stream/issues/47 I am willing to sacrify a little concurrency for raw speed. I do not expect stdinho will be used for large number of clients. But like I said, so far I didn't notice any effect of this, it would be probably more visible with very slow streams.

Reply:
With slow streams it shouldn't matter at all. The data will arrive much slower than any IPC. Slow streams are were blocking could hurt you since the process may block waiting for data while it could be sending more data to the client.

Reply:
My line of thinking is fread will block until it receives specified number of bytes, but I didn't test this.

Reply:
In `bin/stdinho`, add `\Amp\File\filesystem(new \Amp\File\BlockingDriver);` after the `define` and see if there's any noticeable performance difference.

Reply:
Yep, it's pretty much same difference as I reported earlier